tests/test_path.py

```diff
@@ -164,3 +170,62 @@ def test_vsi_path_zip_plus_https():
     """A zip+https:// URLs vsi path is correct (see #1151)"""
     url = 'zip+https://example.com/foo.zip!bar.tif'
     assert vsi_path(parse_path(url)) == '/vsizip/vsicurl/https://example.com/foo.zip/bar.tif'
+
+
+@pytest.mark.parametrize("path, expected_path", [
+    (
+        "netcdf:/input/PLANET_SCOPE_3D.nc:blue",
+        "netcdf:/input/PLANET_SCOPE_3D.nc:blue",
+    ),
+    (
+        "netcdf:https://path/to/file/PLANET_SCOPE_3D.nc?0098y20423%:blue",
+        "netcdf:/vsicurl/https://path/to/file/PLANET_SCOPE_3D.nc?0098y20423%:blue",
+    ),
+    (
+        'netcdf:/vsicurl/"https://path/to/file/PLANET_SCOPE_3D.nc?0098y20423%":blue',
+        'netcdf:/vsicurl/"https://path/to/file/PLANET_SCOPE_3D.nc?0098y20423%":blue',
+    ),
+    (
+        'netcdf:"/vsicurl/https://path/to/file/PLANET_SCOPE_3D.nc?0098y20423%":blue',
+        'netcdf:"/vsicurl/https://path/to/file/PLANET_SCOPE_3D.nc?0098y20423%":blue',
+    ),
+    (
+        'netcdf:/vsicurl/https://path/to/file/PLANET_SCOPE_3D.nc?0098y20423%:blue',
+        'netcdf:/vsicurl/https://path/to/file/PLANET_SCOPE_3D.nc?0098y20423%:blue',
```

Comment:
These are idiomatic GDAL paths again :+1: 

There's another GDAL form, which is something like `/vsicurl?url=https://...`. What would you think about testing for that?